packages/muelu/src/MueCentral/MueLu_VariableContainer.hpp

```diff
@@ -243,31 +251,45 @@ namespace MueLu {
       typedef Teuchos::RCP<Operator> TO;
       typedef Teuchos::RCP<Matrix>   TM;
 
-      const std::string typeTOName = Teuchos::TypeNameTraits<TO>::name();
-      const std::string typeTMName = Teuchos::TypeNameTraits<TM>::name();
-      TEUCHOS_TEST_FOR_EXCEPTION(data_ == NULL, Teuchos::bad_any_cast,
-                                 "Error, cast to type Data<" << typeTOName << "> failed since the content is NULL");
+      if (data_ == NULL) // NVR added guard to avoid unnecessary call to TypeNameTraits<TO>::name(), which calls Teuchos::demangleName(), which can be expensive
```

Comment:
I think you mean `||`, not `&&`.  But even then, this will not do quite the same thing as what I am proposing: in the event that `(data_ != NULL)` and `(data_->type() == typeid(TO))` but `dynamic_cast<Data<TO>*>(data_) != NULL`, there will still be an unnecessary call to `name()` in the argument to the first `TEUCHOS_TEST_FOR_EXCEPTION()`, despite the fact that no exception will be thrown.